tests/main_tests/test_data_check.py

```diff
@@ -0,0 +1,45 @@
+import pathlib
+import sys
+import unittest
+
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+sys.path.append("tests")
+sys.path.append("..src")
+from src.utils.helper_functions.generate_test_data import generate_test_data
+
+
+class TestDataCheck(unittest.TestCase):
+    def test_data_check(self):
+        """
+        check to ensure that the test data is generated correctly
+
+        returns:
+            None
+        """
+
+        file_out_path = "tests/data/test_data.csv"
+        self.df = generate_test_data(file_out_path, 100)
```

Comment:
Consider validating the row length of the test data as an additional test here.

Reply:
added function